File: TD1/echantillonnage.m
%% Donnees standard
freqEchantillon = 44100.0;
pi = 3.141592;
size = 1024;
i = 1:size;

%% Definition du signal
signal.amp = 1.0;
signal.phase = 0.0;
signal.freq = 440.0;

%% Echantillonage
signal.sig = signal.amp * sin(2.0 * pi * signal.freq * i / freqEchantillon + signal.phase);

%% Affichage graphique
close all
figure()
hold on

axis([0 size -1 1]);
plot(signal.sig);

hold off

File: TD1/echantillonnage.c
#include <math.h>

#define PIVALUE 3.141592
#define FECH 44100.0

#define AMP 1.0
#define FREQ 440
#define PHASE 0


void echantillonage () {
	int i=0;
	float temps = 0.0;
	float signal[(int)FECH];


	for (i=0; i<FECH; i++) {
		temps = (float)i/(float)FECH;
		signal[i] = AMP * sin(2.0*PIVALUE * FREQ * temps + PHASE);
	}
}
